operations/mimir/multi-zone.libsonnet

```diff
@@ -86,7 +86,7 @@
   newIngesterZoneStatefulSet(zone, container)::
     local name = 'ingester-zone-%s' % zone;
 
-    self.newIngesterStatefulSet(name, container, with_anti_affinity=false) +
+    self.newIngesterStatefulSet(name, container, with_anti_affinity=!$._config.ingester_allow_multiple_replicas_on_same_node) +
```

Comment:
Doh! This should always be false, because this function injects a different anti-affinity rule below (see `spec+:`). What I meant in the other comment is that we should inject the custom anti-affinity rule only if `ingester_allow_multiple_replicas_on_same_node: false`.

Reply:
I should have paid more attention to that, thank you.

This should be the fix: https://github.com/grafana/mimir/pull/1581/commits/3d534a740406c72f09a29195c3bf2e4122f7e60f